### PR TITLE
feat(coco client): Improve output & show progress

### LINKED ISSUE #186: coco client: print sth while waiting
Apparently users are confused by the coco client seemingly "hanging" when it's waiting for a response. Especially if that takes a long time due to a high coco queue fill level.

One useful information to print for the user (e.g. every 30s) is the queue fill level (get it from the prometheus web server coco is running)/

### PR DESCRIPTION
**Pull request by nritsche**
Closes #186

### CODE REVIEW

File: coco/endpoint.py [hunk 1]
```diff
@@ -551,16 +564,74 @@ def client_send_request(host, port, args):
...
+                    queue_size = asyncio.create_task(
+                        print_queue_size(metric_request_count)
+                    )
+                    done, pending = await asyncio.wait(
+                        {result, queue_size}, return_when=asyncio.FIRST_COMPLETED
+                    )
+                    if result in done:
+                        queue_size.cancel()
+                        print("/n")
+                        break
```
Comment (anjakefala) [inline coco/endpoint.py]: Why do we want to break out of the loop for `print_queue_size` completing?

File: coco/endpoint.py [hunk 2]
```diff
@@ -551,16 +564,74 @@ def client_send_request(host, port, args):
...
+                            result = {"Error": await resp.text()}
+                except BaseException as e:
+                    return f"coco-client: Sending request failed: {e}"
+                else:
+                    return result
+
+        async def request_and_wait():
+            result = asyncio.create_task(send_request())
+            if not args.silent:
+                metric_request_count = 0
```
Comment (nritsche) [inline coco/endpoint.py]: TODO: explain what is going on here

File: coco/endpoint.py (hunk 1 above)
Comment (anjakefala) [reply coco/endpoint.py]: I got confused about the various tasks!

File: coco/endpoint.py [hunk 3]
```diff
@@ -551,16 +564,74 @@ def client_send_request(host, port, args):
...
+                while True:
+                    metric_request_count = metric_request_count + 1
+                    queue_size = asyncio.create_task(
+                        print_queue_size(metric_request_count)
+                    )
+                    done, pending = await asyncio.wait(
+                        {result, queue_size}, return_when=asyncio.FIRST_COMPLETED
```
Comment (nritsche) [inline coco/endpoint.py]: TODO: rename `result` to something more meaningful (e.g. `main_request`)

File: coco/metric.py [hunk 4]
```diff
@@ -48,3 +52,34 @@ def start_metrics_server(port, callbacks=None, addr=""):
...
+    -------
+    Value of the metric
+    """
+    async with aiohttp.ClientSession() as session:
+        async with session.get(f"http://{host}:{port}") as resp:
+            resp.raise_for_status()
+            metrics = await resp.text()
+    for family in text_string_to_metric_families(metrics):
+        if family.name == name:
+            assert len(list(family.samples)) == 1
```
Comment (anjakefala) [inline coco/metric.py]: Why is it important that their is only 1 sample?

File: coco/metric.py [hunk 5]
```diff
@@ -48,3 +52,34 @@ def start_metrics_server(port, callbacks=None, addr=""):
...
+            for sample in family.samples:
+                return sample[2]
```
Comment (anjakefala) [inline coco/metric.py]: Why is the `2` element in a sample?

File: coco/endpoint.py [hunk 6]
```diff
@@ -531,15 +541,18 @@ def client_send_request(host, port, args):
...
         type : str
             HTTP request type.
         data : json
             JSON data.
         args : :class:`argparse.Namespace`
-            Namespace populated by argparse. For now only may contain the report type:
-            `report : str`.
+            Namespace populated by argparse. May contain the report type
+            (`report : str`) and the refresh time for the client  in seconds
+            ('client-refresh-time' : int)
```
Comment (anjakefala) [inline coco/endpoint.py]: And `('silent' : boolean) =)

**Comment by anjakefala** [comment]
I had small clarification questions that we may want to add documentation for, but otherwise this looks wonderful! :blush: :tada:

File: coco/metric.py (hunk 4 above)
Comment (nritsche) [reply coco/metric.py]: I wouldn't know why there would be more than 1...

Comment (nritsche) [reply coco/metric.py]: Do you?

Comment (anjakefala) [reply coco/metric.py]: Why did they call the attribute `samples` .-.

**Pull request merged**